dataclients/kubernetes/endpointslices_test.go

```diff
@@ -1,51 +1,91 @@
 package kubernetes
 
 import (
-	"net/url"
-	"strconv"
 	"testing"
 
-	"github.com/zalando/skipper/dataclients/kubernetes/definitions"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestTargets(t *testing.T) {
-	want := "http://10.0.0.1:8080"
-	u, err := url.Parse(want)
-	if err != nil {
-		t.Fatalf("Failed to parse: %v", err)
-	}
-	port, err := strconv.Atoi(u.Port())
-	if err != nil {
-		t.Fatalf("Failed to parse: %v", err)
-	}
+func TestAddresses(t *testing.T) {
+	assert.Equal(t, []string{"10.0.0.1:8080"}, (&skipperEndpointSlice{
+		Endpoints: []*skipperEndpoint{
+			{
+				Address: "10.0.0.1",
+				Zone:    "zone-1",
+			},
+		},
+		Ports: []*endpointSlicePort{
+			{
+				Name:     "main",
+				Port:     8080,
+				Protocol: "TCP",
+			},
+		},
+	}).addresses())
 
-	ses := &skipperEndpointSlice{
-		Meta: &definitions.Metadata{
-			Namespace: "ns1",
-			Name:      "a-slice",
+	assert.Equal(t, []string{"10.0.0.1:8080", "10.0.0.1:8081"}, (&skipperEndpointSlice{
```

Comment:
This looks wrong to me, because if you have a multi port service this could lead to mixed up ports as far as I understand.

Reply:
yes, and this is how it is implemented for endpoints. I.e. this function is usable for single-port services

Another option is to return only IPs without port and make port explicit (I think we even have a flag for it) or get first port from the service explicitly.

Reply:
Decided to return addresses without ports